packages/middleware-compression/src/resolveCompressionConfig.ts

```diff
@@ -5,7 +5,9 @@ import { CompressionInputConfig, CompressionResolvedConfig } from "./configurati
 /**
  * @internal
  */
-export const resolveCompressionConfig = <T>(input: T & CompressionInputConfig): T & CompressionResolvedConfig => ({
+export const resolveCompressionConfig = <T>(
+  input: T & Required<CompressionInputConfig>
+): T & CompressionResolvedConfig => ({
   ...input,
   disableRequestCompression: normalizeProvider(input.disableRequestCompression),
   requestMinCompressionSizeBytes: async () => {
```

Comment:
nit: is it safer to guard against `undefined` values for `disableRequestCompression` and `requestMinCompressionSizeBytes`, and not make `CompressionInputConfig` required?

Reply:
There won't be undefined values when calling `resolveCompressionConfig` as runtimeConfig resolution ensure that it's defined. If it doesn't, the client compilation should fail like it's happening in protocol tests.

Also, the [Required](https://www.typescriptlang.org/docs/handbook/utility-types.html) doesn't allow undefined.
[Playground Link](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgApQPYAcDOyDeAUMsnAPwBcyIArgLYBG0A3Mcg5cjmFKAOasAvsUIIMIbsgwMAVlXTY8AXgKkqAVmSDWyUeMnSZAJioAlCAEcawKBAAmAHgW4AfMhX41ydQBp2VGhA7CBhQey1mIA)